packages/grid/_modules_/grid/components/containers/GridRootStyles.ts

```diff
@@ -222,6 +222,10 @@ export const GridRootStyles = styled('div', {
           },
         },
       },
+      [`&.${gridClasses['row--last']}:not(.${gridClasses['row--lastBeforeEmpty']}) .${gridClasses.cell}`]:
```

Comment:
I found the purpose of `row--lastBeforeEmpty` hard to understand. How about adding a class to `MuiDataGrid-virtualScrollerContent` to signal that its content doesn't extend to the full height of the grid. You'll need to wait for #3821 to able to do it. In DataGridPro the calculation is already being done, we only need to add the class name. 

https://github.com/mui/mui-x/blob/996f727467311f20e999b8db3d1331e3526594fd/packages/grid/x-data-grid-pro/src/DataGridProVirtualScroller.tsx#L249-L250

Once you have the class name added you can create a selector like `.MuiDataGrid-root:not(.MuiDataGrid-autoHeight) .MuiDataGrid-virtualScroller:not(.MuiDataGrid-virtualScrollerContent--newClassAdded) .MuiDataGrid-row--last`.

Reply:
That will work as well. I can put the on hold for now for the PR until the other one is merged.

Reply:
Updated